src/Host/Client/Impl/Program.cs

```diff
@@ -190,13 +167,32 @@ class MaxLoggingPermissions : ILoggingPermissions {
             public LogVerbosity MaxVerbosity => LogVerbosity.Traffic;
         }
 
-        class AppConstants : IApplicationConstants {
-            public void Initialize() { }
-            public string ApplicationName => "Microsoft.R.Host.Client";
-            public IntPtr ApplicationWindowHandle => IntPtr.Zero;
-            public uint LocaleId => 1033;
-            public string LocalMachineHive => null;
-            public UIColorTheme UIColorTheme => UIColorTheme.Light;
+        class CoreShell : ICoreShell {
+            private readonly ServiceManager _serviceManager;
+
+            public CoreShell(IActionLog log) {
+                ThreadId = Thread.CurrentThread.ManagedThreadId;
+                _serviceManager = new ServiceManager();
+                _serviceManager
+                    .AddService(new MaxLoggingPermissions())
+                    .AddService(log);
+            }
+
+            public string ApplicationName => "Program";
+            public int LocaleId => 1033;
+            public IServiceContainer Services => _serviceManager;
+
+            public bool IsUnitTestEnvironment => false;
+            public int ThreadId { get; }
+
+#pragma warning disable 67
+            public event EventHandler<EventArgs> Started;
+            public event EventHandler<EventArgs> Terminating;
+            public event EventHandler<EventArgs> Idle;
+#pragma warning restore 67
+
+            public void DispatchOnUIThread(Action action) => action();
```

Comment:
Not part of this change, but we probably should move `DispatchOnUIThread` into `IMainThread`